src/context/src/input.rs

```diff
@@ -1,34 +1,71 @@
 //! The input handler for the game engine
 
-use std::collections::HashSet;
+use std::collections::hash_map::{Entry, HashMap};
 use event::{ElementState, EngineEvent, Event, VirtualKeyCode};
 
 pub struct InputHandler {
-    keys_down: HashSet<VirtualKeyCode>,
+    pressed_keys: HashMap<VirtualKeyCode, bool>,
 }
 
 impl InputHandler {
     /// Create a new InputHandler
     pub fn new() -> InputHandler {
         InputHandler {
-            keys_down: HashSet::new(),
+            pressed_keys: HashMap::new(),
         }
     }
 
     /// Update the input handler with new engine events
     pub fn update(&mut self, events: &[EngineEvent]) {
         for event in events {
             match event.payload {
-                Event::KeyboardInput(ElementState::Pressed, _, Some(key_code)) => { self.keys_down.insert(key_code); }
-                Event::KeyboardInput(ElementState::Released, _, Some(key_code)) => { self.keys_down.remove(&key_code); }
-                Event::Focused(false) => self.keys_down.clear(),
+                Event::KeyboardInput(ElementState::Pressed, _, Some(key_code)) => {
+                    match self.pressed_keys.entry(key_code) {
+                        Entry::Occupied(mut entry) => {
+                            entry.insert(false);
```

Comment:
I think this should actually be a no-op. If there is a consecutive `Pressed` event (e,g. windows' repeat events) but `key_once` hasn't been called yet, the entry should stay as `true`.